main.js

```diff
@@ -0,0 +1,94 @@
+// Clase Producto
+class Producto {
+  constructor(nombre, precio) {
+    this.nombre = nombre;
+    this.precio = precio;
+  }
+
+  calcularSubtotal(cantidad) {
+    return this.precio * cantidad;
+  }
+}
+
+// Clase Carrito
+class Carrito {
+  constructor() {
+    this.productos = [];
+  }
+
+  agregarProducto(producto, cantidad) {
+    const subtotal = producto.calcularSubtotal(cantidad);
+    this.productos.push({
+      nombre: producto.nombre,
+      precio: producto.precio,
+      cantidad,
+      subtotal
+    });
+  }
+
+  calcularTotal() {
+    return this.productos.reduce((acc, item) => acc + item.subtotal, 0);
+  }
+
+  mostrarResumen() {
+    let mensaje = "🛒 RESUMEN DEL PEDIDO 🛒\n\n";
+    this.productos.forEach(p => {
+      mensaje += `${p.nombre} x${p.cantidad} = $${p.subtotal}\n`;
+    });
+    mensaje += `\nTOTAL: $${this.calcularTotal()}`;
+    alert(mensaje);
+  }
+}
+
+// Lista de productos disponibles
+const listaProductos = [
+  new Producto("Alimento para perro", 3500),
+  new Producto("Juguete de goma", 1200),
+  new Producto("Rascador para gatos", 4200),
+  new Producto("Collar con luz", 2800)
+];
+
+// Mostrar productos al usuario
+function mostrarCatalogo() {
+  let mensaje = "Productos disponibles:\n";
+  listaProductos.forEach((p, i) => {
+    mensaje += `${i + 1}. ${p.nombre} - $${p.precio}\n`;
+  });
+  alert(mensaje);
+}
+
+// Simulador de compra
+function simuladorDeCompra() {
+  const carrito = new Carrito();
+  let seguir = true;
+
+  alert("Bienvenido al simulador de PetShop Online 🐾");
+
+  while (seguir) {
+    mostrarCatalogo();
+    const opcion = parseInt(prompt("Ingresá el número del producto que querés agregar:"));
+
+    if (opcion >= 1 && opcion <= listaProductos.length) {
```

Comment:
Si el prompt devuelve null, parseInt puede arrojar NaN. Estás manejando esto parcialmente, pero podrías validarlo de forma más robusta.
```suggestion
const input = prompt("Ingresá el número del producto que querés agregar:");
const opcion = parseInt(input);

if (!isNaN(opcion) && opcion >= 1 && opcion <= listaProductos.length) {
  ...
}
```

Reply:
Si el usuario presiona “Cancelar” en un prompt, se obtiene null, y parseInt(null) da NaN.